.github/workflows/ci-cd.yml

```
@@ -0,0 +1,754 @@
+      ${{
+          (
+            (
+              needs.pre-setup.outputs.release-requested == 'true' &&
+              !toJSON(github.event.inputs.YOLO)
+            ) ||
+            contains(matrix.py, '-dev')
+          ) && true || false
+      }}
+
+    env:
+      MYSQL_ROOT_PASSWORD: rootpw
+      PY_COLORS: 1
+
+    services:
+      mysql:
+        image: "${{ join(matrix.db, ':') }}"
+        ports:
+        - 3306:3306
+        volumes:
+        - "/tmp/run-${{ join(matrix.db, '-') }}/:/socket-mount/"
+        options: '--name=mysqld'
+        env:
+          MYSQL_ROOT_PASSWORD: rootpw
+
+    steps:
+    - name: Setup Python ${{ matrix.py }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.py }}
+
+    - name: Figure out if the interpreter ABI is stable
+      id: py-abi
+      run: |
+        from sys import version_info
+        is_stable_abi = version_info.releaselevel == 'final'
+        print(
+            '::set-output name=is-stable-abi::{is_stable_abi}'.
+            format(is_stable_abi=str(is_stable_abi).lower())
+        )
+      shell: python
+
+    - name: >-
+        Calculate Python interpreter version hash value
+        for use in the cache key
+      if: fromJSON(steps.py-abi.outputs.is-stable-abi)
+      id: calc-cache-key-py
+      run: |
+        from hashlib import sha512
+        from sys import version
+        hash = sha512(version.encode()).hexdigest()
+        print('::set-output name=py-hash-key::{hash}'.format(hash=hash))
+      shell: python
+
+    - name: Get pip cache dir
+      if: fromJSON(steps.py-abi.outputs.is-stable-abi)
+      id: pip-cache-dir
+      run: >-
+        echo "::set-output name=dir::$(python -m pip cache dir)"
+
+    - name: Set up pip cache
+      if: fromJSON(steps.py-abi.outputs.is-stable-abi)
+      uses: actions/cache@v2.1.7
+      with:
+        path: ${{ steps.pip-cache-dir.outputs.dir }}
+        key: >-
+          ${{ runner.os }}-pip-${{
+          steps.calc-cache-key-py.outputs.py-hash-key }}-${{
+          needs.pre-setup.outputs.cache-key-files }}
+        restore-keys: |
+          ${{ runner.os }}-pip-${{
+              steps.calc-cache-key-py.outputs.py-hash-key
+          }}-
+          ${{ runner.os }}-pip-
+
+    - name: Update pip
+      run: >-
+        python -m
+        pip install
+        --user
+        pip
+
+    - name: Grab the source from Git
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.inputs.release-commitish }}
+
+    - name: Download all the dists
+      uses: actions/download-artifact@v2
+      with:
+        name: python-package-distributions
+        path: dist/
+
+    - name: Install dependencies
+      run: >-
+        python -m
+        pip install
+        --user
+        --requirement requirements-dev.txt
+
+    - name: Install previously built wheel
+      run: >-
+        python -m
+        pip install
+        --no-cache-dir
+        --user
+        --no-index
```

I don't think these are needed. With an explicit file path, the wheel we need will get installed and the deps could be received from wherever they are.
```suggestion
        --user
```